catalog/app/containers/Bucket/requests.js

```diff
@@ -218,8 +218,28 @@ export const bucketStats = async ({ req, s3, bucket, overviewUrl }) => {
   throw new Error('Stats unavailable')
 }
 
+// key is array of keys: string[]
+const ensureObjectIsPresentInCollection = async ({ s3, bucket, key }) => {
+  if (!key.length) return null
+
+  const path = R.last(key)
+  const fileExists = await ensureObjectIsPresent({
+    s3,
+    bucket,
+    key: path,
+  })
+
+  return (
+    fileExists ||
+    (await ensureObjectIsPresentInCollection({ s3, bucket, key: R.init(key) }))
+  )
+}
+
 const fetchFileVersioned = async ({ s3, bucket, path, version }) => {
-  const versionExists = await ensureObjectIsPresent({
+  const isPresent = Array.isArray(path)
```

Comment:
we could just normalize path (convert it to a list) and use `ensureObjectIsPresentInCollection`